test/integration_test/basic_test.go

```diff
@@ -439,67 +416,56 @@ func BasicUpgradeOperator(tc *types.TestCase) func(*testing.T) {
 	}
 }
 
-func InstallWithTelemetry(tc *types.TestCase) func(*testing.T) {
+// BasicTelemetryRegression test includes the following steps:
+// 1. Deploy PX and validate Telemetry is not enabled by default
+// 2. Enable Telemetry and validate all its components got deployed
+// 3. Disable Telemetry and validate its components got deleted
+// 4. Delete StorageCluster and validate it got successfully removed (including px-telemetry-certs secret)
+func BasicTelemetryRegression(tc *types.TestCase) func(*testing.T) {
 	return func(t *testing.T) {
+		var err error
 		testSpec := tc.TestSpec(t)
 		cluster, ok := testSpec.(*corev1.StorageCluster)
 		require.True(t, ok)
 
-		testInstallWithTelemetry(t, cluster)
-	}
-}
+		// Create StorageCluster
+		cluster = ci_utils.DeployAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Validate Telemetry is not enabled by default
+		logrus.Info("Validate Telemetry is not enabled by default")
+		if cluster.Spec.Monitoring != nil {
+			if cluster.Spec.Monitoring.Telemetry != nil {
+				if cluster.Spec.Monitoring.Telemetry.Enabled {
+					require.False(t, cluster.Spec.Security.Enabled, "failed to validate default Telemetry status: expected: false, actual: %v", cluster.Spec.Monitoring.Telemetry.Enabled)
```

Comment:
nit: by default the field is nil, we expect it to be nil or false, so if it's enabled, we can call `require.FailNow()`. but this should work fine.

Reply:
If this is fine I will just leave it as is